engine/src/main/java/org/camunda/bpm/engine/impl/cfg/CommandChecker.java

```diff
@@ -271,6 +271,11 @@ public interface CommandChecker {
    */
   void checkDeleteTask(TaskEntity task);
 
+  /**
+   * Check if it is allowed to update a task
+   */
+  void checkUpdateTask(TaskEntity task);
+
```

Comment:
That makes sense for actions for comments using taskId.

How about update/delete comments using processInstanceId ? 

Can we use this? 

https://github.com/camunda/camunda-bpm-platform/blob/c307a42aedaf627cd6ece79e8dc837343e0d59e3/engine/src/main/java/org/camunda/bpm/engine/impl/cfg/auth/AuthorizationCommandChecker.java#L256

Reply:
> That makes sense for actions for comments using taskId.
> 
> How about update/delete comments using processInstanceId ?
> 
> Can we use this?
> 
> https://github.com/camunda/camunda-bpm-platform/blob/c307a42aedaf627cd6ece79e8dc837343e0d59e3/engine/src/main/java/org/camunda/bpm/engine/impl/cfg/auth/AuthorizationCommandChecker.java#L256

Digging more into it, for me using checkTaskAssign makes more sense on this on as well.  Let me know if you think otherwise.

Reply:
@prajwolbhandari1, by bad for the wrong reference. `#checkTaskWork()` should be used, it covers both API using task and process instance ids.
https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/main/java/org/camunda/bpm/engine/impl/cfg/auth/AuthorizationCommandChecker.java#L676C15-L676C28

Reply:
> @prajwolbhandari1, by bad for the wrong reference. `#checkTaskWork()` should be used, it covers both API using task and process checks. https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/main/java/org/camunda/bpm/engine/impl/cfg/auth/AuthorizationCommandChecker.java#L676C15-L676C28

No worries Yana. I will make these changes. Thank you.